source/FreeRTOS_IP_Utils.c

```diff
@@ -220,6 +220,10 @@ static NetworkBufferDescriptor_t * prvPacketBuffer_to_NetworkBuffer( const void
     else
     {
         /* Obtain the network buffer from the zero copy pointer. */
+
+        /* the conversion here does not cause a loss of data, as uintptr_t fits a
+         * pointer type on the system */
+        /* coverity[misra_c_2012_rule_11_1_violation] */
         uxBuffer = ipPOINTER_CAST( uintptr_t, pvBuffer );
```

Comment:
Something like:
```
uint8_t * pcBufferOffset = &( ( ( uint8_t * ) pvBuffer )[ 0 -uxOffset - ipBUFFER_PADDING ] );
```

Would also suffice.

Reply:
About the following:
~~~
uint8_t * pcBufferOffset = &( ( ( uint8_t * ) pvBuffer )[ 0 - uxOffset - ipBUFFER_PADDING ] );
~~~
This is the transformation to the beginning of the allocated buffer.

In my experience, coverity want to see such a transformation in smaller steps:

uint8_t * pcBufferOffset = ( uint8_t * ) pvBuffer;
pcBufferOffset = &( pcBufferOffset [ 0 - ( uxOffset + ipBUFFER_PADDING ) ] );

or even more steps.

Reply:
Agree. Similar to the change https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/491/files

Reply:
Rule 11.1 refers to function pointers, but this is not a function pointer.